### PR TITLE
Add follow organization and fix the logic of following page

### PR DESCRIPTION
**Pull request by yp05327**
image
image

Maybe we can fix user card tmpl in #24319?
Or maybe a list is better here
image

### CODE REVIEW

File: models/user/user.go [hunk 1]
```diff
@@ -346,7 +346,6 @@ func GetUserFollowing(ctx context.Context, u, viewer *User, listOptions db.ListO
 		Select("`user`.*").
 		Join("LEFT", "follow", "`user`.id=follow.follow_id").
 		Where("follow.user_id=?", u.ID).
-		And("`user`.type=?", UserTypeIndividual).
```
Comment (techknowlogick) [inline models/user/user.go:349]: could you change this condition to be `type in (0, 1)` as we now have more types than just users and orgs?

**Comment by yp05327** [comment]
3 following but 2 users displayed:
image

There's a bug in `isUserVisibleToViewerCond`

**Comment by yp05327** [comment]
The conds in `isUserVisibleToViewerCond` confuse me.

Maybe this cond means `user.id In viewer.ID` ?
image

And why `INNER JOIN`?
image

**Comment by yp05327** [comment]
As `isUserVisibleToViewerCond` added in #20220, we need to dynamically calculate the count of `followers` and `following`, 
maybe `User.NumFollowers` and `User.NumFollowing` is unnecessary now.

**Comment by silverwind** [comment]
Can you make the button just plain `basic` button without color? I think that red/green draws way to much attention for such an action.

**Comment by yp05327** [comment]
> Can you make the button just plain `basic` button without color? I think that red/green draws way to much attention for such an action.

Done. Also removed the icon.
Now it is similar to github.
image

**Comment by yp05327** [comment]
@lunny 
I changed my mind. 
Remove `User.NumFollowers` and `User.NumFollowing` should be in another PR.
Because this change will remove many codes which are not related to this PR. 
So do this in anohter PR is better.

**Comment by silverwind** [comment]
Pushed a tweak.

**Review by silverwind — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**